src/HotChocolate/Core/src/Types/Types/Relay/Attributes/NodeAttribute.cs

```diff
@@ -92,7 +92,7 @@ public class NodeAttribute : ObjectTypeDescriptorAttribute
             }
             else if (NodeResolver is not null)
             {
-                MethodInfo? method = type.GetMethod(NodeResolver);
+                MethodInfo? method = type.GetMethod(NodeResolver, BindingFlags.Instance | BindingFlags.Static | BindingFlags.Public | BindingFlags.FlattenHierarchy);
```

Comment:
we need to check for sideffect for FlattenHirarchy. How do you do overrides with static methods and such